x-pack/packages/security-solution/upselling/sections/entity_analytics.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useCallback } from 'react';
+import {
+  EuiCard,
+  EuiIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiText,
+  EuiButton,
+  EuiTextColor,
+} from '@elastic/eui';
+import { useNavigation } from '@kbn/security-solution-navigation';
+import styled from '@emotion/styled';
+import * as i18n from '../pages/translations';
+
+const StyledEuiCard = styled(EuiCard)`
+  span.euiTitle {
+    max-width: 540px;
+    display: block;
+    margin: 0 auto;
+  }
+`;
+
+export const EntityAnalyticsUpsellingSection = memo(
+  ({
+    requiredLicense,
+    requiredProduct,
+    subscriptionUrl,
+  }: {
+    requiredLicense?: string;
+    requiredProduct?: string;
+    subscriptionUrl?: string;
+  }) => {
```

Comment:
Since this is implemented in the package, it would be better if the offering-specific knowledge for serverless (product) and ess (license) was owned by the callers (serverless or ess plugins). This component should implement the common parts. 
Consider changing the props to:

```suggestion
export const EntityAnalyticsUpsellingSection = memo(
  ({
    upgradeMessage,
    upgradeTo,
    upgradeHref,
  }: {
    upgradeMessage: string;
    upgradeTo: string;
    upgradeHref: string;
  }) => {
```

Moving offering-specific translations to the offering plugins accordingly.